plugins/e57/io/E57Writer.cpp

```diff
@@ -218,12 +256,37 @@ void E57Writer::done(PointTableRef table)
     if (Utils::contains(m_dimensionsToWrite, "intensity"))
     {
         // found intensity info
-        e57::StructureNode colorbox = e57::StructureNode(*m_imageFile);
-        colorbox.set("intensityMinimum",
-                     e57::IntegerNode(*m_imageFile, 0));
-        colorbox.set("intensityMaximum",
-                     e57::IntegerNode(*m_imageFile, m_chunkWriter->getIntensityLimit()));
-        m_scanNode->set("intensityLimits", colorbox);
+        e57::StructureNode intensityBox = e57::StructureNode(*m_imageFile);
+        intensityBox.set("intensityMinimum",
+                         e57::IntegerNode(*m_imageFile, 0));
+        intensityBox.set(
+            "intensityMaximum",
+            e57::IntegerNode(*m_imageFile, m_chunkWriter->getIntensityLimit()));
+        m_scanNode->set("intensityLimits", intensityBox);
+    }
+
+    if (Utils::contains(m_dimensionsToWrite, "classification"))
+    {
+        // found classification info
+        e57::StructureNode classificationBox = e57::StructureNode(*m_imageFile);
+        classificationBox.set("classificationMinimum",
+                              e57::IntegerNode(*m_imageFile, 0));
+        classificationBox.set(
+            "classificationMaximum",
+            e57::IntegerNode(*m_imageFile, 255));
+        m_scanNode->set("classificationLimits", classificationBox);
+    }
+
+    for (auto extradim = m_extraDims->begin(); extradim != m_extraDims->end();
+            ++extradim)
+    {
+        e57::StructureNode classificationBox = e57::StructureNode(*m_imageFile);
+        classificationBox.set(extradim->m_name+"Minimum",
+                              e57::IntegerNode(*m_imageFile, std::floor(extradim->m_min)));
```

Comment:
It might be better to set that as a FloatNode. Like that you can support any kind of data

Reply:
I don't think classification will float in any case.

Reply:
Ok.

Reply:
Done